pkg/handlers/clusterlogging/handler.go

```diff
@@ -37,38 +40,66 @@ func NewHandlers(opts *config.Options) []handlers.RequestHandler {
 }
 
 func (ext *handler) Process(req *http.Request, context *handlers.RequestContext) (*http.Request, error) {
-	name := context.UserName
-	if context.IsWhiteListed(name) || ext.hasInfraRole(context) {
-		log.Debugf("Skipping additional processing, %s is whitelisted or has the infra role", name)
-		return req, nil
+
+	log.Infof("Received context: UserName: %v", context.UserName)
+	log.Infof("Received context: Roles: %v", context.Roles)
+
+	// if we are getting a .kibana/config url...
+	if ext.isKibanaIndex(*req.URL) {
+		kibanaVersion := ext.getKibanaVersion(*req.URL)
+		if kibanaVersion != "" {
+			log.Infof("Received request: kibana ver: %v", kibanaVersion)
+		}
+
+		log.Infof("Received request: Header: %v", req.Header)
+
+		// use ES client and set headers for Index request (WithHeaders) from req...
+		// we let the security plugin correctly choose the tenant -- so we only seed as required
+		// only seed infra and audit if they are an admin role though...
+
+		// to create index pattern:
```

Comment:
Will we always create or will we fetch... if not exists... create?

Reply:
since the `Index` api call to ES will create or update it I think we can save ourselves a call and just do the create here. the default pattern will need the fetch first so we don't change a user's default indexpattern if they've updated it from what we set -- we just need to ensure they have a default.